RadioDns-PlugIt/SPI/modules/aws_spi.py

```diff
@@ -42,26 +42,48 @@ def on_pi_resource_changed(self, event_name, station):
             start_of_the_week = datetime.datetime.combine(today - datetime.timedelta(days=today.weekday()),
                                                           datetime.time())
 
-            for i in range(0, 7):
+            for i in range(0, 14): # RadioEPG 1.1
                 start_date = (start_of_the_week + datetime.timedelta(days=i)).strftime("%Y%m%d")
-                contents = SPI.utils.generate_pi_file(start_date, station=station)
+                contents = SPI.utils.generate_pi_file(start_date, "/templates/radioepg/schedule/xml1.html", station=station)
                 if not contents:
                     continue
 
                 filtered = Channel.query.filter(Channel.station_id == station.id and Channel.type_id in ["fm", "dab"])
                 for channel in filtered:
-                    for w in range(0, 2):
-                        try:
-                            self.upload_file(
-                                AWSSPI.get_static_bucket_pi_filename(
-                                    channel.service_identifier,
-                                    (start_of_the_week + datetime.timedelta(weeks=w, days=i)).strftime("%Y%m%d")
-                                ),
-                                contents,
-                                station.id,
-                            )
-                        except AttributeError as e:
+                    try:
+                        self.upload_file(
+                            AWSSPI.get_static_bucket_pi_filename(
+                                1,
+                                channel.service_identifier,
+                                start_date.strftime("%Y%m%d")
+                            ),
+                            contents,
+                            station.id,
+                        )                         
+                    except AttributeError as e:
                             print("[WARN][AWS] in pi resource changed handler", e)
+
+            for i in range(0, 14): # SPI 3.1
+                start_date = (start_of_the_week + datetime.timedelta(days=i)).strftime("%Y%m%d")
+                contents = SPI.utils.generate_pi_file(start_date, "/templates/radioepg/schedule/xml3.html", station=station)
+                if not contents:
+                    continue
+
+                filtered = Channel.query.filter(Channel.station_id == station.id and Channel.type_id in ["fm", "dab"])
+                for channel in filtered:
+                    try:
+                        self.upload_file(
+                            AWSSPI.get_static_bucket_pi_filename(
+                                1,
```

Comment:
What about version 3=

Reply:
Its in there, both support for routes and generating.